components/ILIAS/Mail/classes/class.ilAccountMail.php

```diff
@@ -296,13 +300,13 @@ public function replacePlaceholders(string $a_string, ilObjUser $a_user, array $
             $this->http->wrapper()->query()->retrieve('target', $this->refinery->kindlyTo()->string()) !== ''
         ) {
             $target = $this->http->wrapper()->query()->retrieve('target', $this->refinery->kindlyTo()->string());
-            $tarr = explode('_', (string) $target);
-            if ($this->repositoryTree->isInTree((int) $tarr[1])) {
+            $tarr = explode("_", $target);
+            if ($tree->isInTree((int) $tarr[1])) {
                 $obj_id = ilObject::_lookupObjId((int) $tarr[1]);
                 $type = ilObject::_lookupType($obj_id);
-                if ($type === $tarr[0]) {
+                if ($type == $tarr[0]) {
                     $replacements["TARGET_TITLE"] = ilObject::_lookupTitle($obj_id);
-                    $replacements["TARGET"] = ILIAS_HTTP_PATH . '/goto.php?client_id=' . CLIENT_ID . '&target=' . $target;
+                    $replacements["TARGET"] = ILIAS_HTTP_PATH . "/goto.php?client_id=" . CLIENT_ID . "&target=" . $_GET["target"];
```

Comment:
Do we really need this change? Is it related to the feature? Why do we introduce a new access to `$_GET` here?